dev-docs/CMakeLists.txt

```diff
@@ -7,23 +7,14 @@ set(DOCTREE_DIR ${CMAKE_CURRENT_BINARY_DIR}/doctree)
 
 # We try to execute Sphinx directly through python -m to avoid problems with the
 # startup scripts on Windows. They are not always reliable as they can have
-# hardcoded paths in them. However, older versions of Sphinx dont't allow python
-# -m execution. Assume we are running on Linux and `sphinx-build` is available
-# in these cases
-if(EXISTS ${SPHINX_PACKAGE_DIR}/__main__.py)
-  add_custom_target(dev-docs-${BUILDER}
-                    COMMAND ${Python_EXECUTABLE} -m sphinx -w dev_docs_warnings.txt
-                            -W --keep-going -b ${BUILDER} -d
-                            ${DOCTREE_DIR} ${CMAKE_CURRENT_LIST_DIR}/source
-                            ${OUT_DIR}
-                    COMMENT "Building html developer documentation")
-else()
-  add_custom_target(dev-docs-${BUILDER}
-                    COMMAND sphinx-build -w dev_docs_warnings.txt
-                            -W --keep-going -b ${BUILDER} -d ${DOCTREE_DIR}
-                            ${CMAKE_CURRENT_LIST_DIR}/source ${OUT_DIR}
-                    COMMENT "Building html developer documentation")
-endif()
+# hardcoded paths in them.
+add_custom_target(dev-docs-${BUILDER}
+                  COMMAND ${Python_EXECUTABLE} -m ${SPHINX_MAIN} ${SPHINX_NO_COLOR}
+                          -w dev_docs_warnings.txt
+                          -W --keep-going -b ${BUILDER}
```

Comment:
I'm getting an error in my CentOS 7 docker container:
```
[1/1] Building html developer documentation
FAILED: dev-docs/CMakeFiles/dev-docs-html
cd /mantid_build/dev-docs && /usr/bin/python3.6 -m sphinx.__init__ -w dev_docs_warnings.txt -W --keep-going -b html -d /mantid_build/dev-docs/doctree /mantid_src/dev-docs/source /mantid_build/dev-docs/html
Error: option --keep-going not recognized
```
It looks like this `--keep-going` argument only got added in [Sphinx v2.0](https://github.com/sphinx-doc/sphinx/commit/925bc187eacbc0fbdd2c56f360a040a23cb13145) so I think we need to similar protection to `--no-color`